### PR TITLE
Add histogram features

### LINKED ISSUE #488: cumulative and histnorm options
These aren't in the current workspace but are really useful and should be in this version :)

### LINKED ISSUE #495: Box plot weirdness
If I select box plot, x and y, then click on style/traces, the display changes and clicking back to create/traces shows that x is empty

### PR DESCRIPTION
**Pull request by dmt0**
Closes #495
Closes #488 
Also fixed a small display issue when there's no margin at bottom of panel

### CODE REVIEW

File: src/default_panels/GraphCreatePanel.js [hunk 1]
```diff
@@ -51,6 +64,16 @@ const GraphCreatePanel = (props, {localize: _}) => {
           }}
           attr="z"
         />
+        <HistogramInfoVertical>
+          {_(
+            'Note: in vertical orientation X values are used for bins and Y values for weights.'
```
Comment (VeraZab) [inline src/default_panels/GraphCreatePanel.js]: just maybe a `,` after, in vertical orientation `,`

File: src/default_panels/GraphCreatePanel.js [hunk 2]
```diff
@@ -51,6 +64,16 @@ const GraphCreatePanel = (props, {localize: _}) => {
...
+          )}
+        </HistogramInfoVertical>
+        <HistogramInfoHorizontal>
+          {_(
+            'Note: in horizontal orientation Y Values are used for bins and X values for weights.'
```
Comment (VeraZab) [inline src/default_panels/GraphCreatePanel.js]: same

**Comment by VeraZab** [comment]
seems good to me 💃

File: src/components/fields/derived.js [hunk 3]
```diff
@@ -144,66 +145,63 @@ export const ContourNumeric = connectToContainer(UnconnectedNumeric, {
...
-        y: context.container.x,
-        ysrc: context.container.xsrc,
-        ybins: context.container.xbins,
-      });
+      context.container.orientation === 'h';
+    return plotProps;
+  },
+});
+
+export const ColorscaleNumeric = connectToContainer(UnconnectedNumeric, {
```
Comment (nicolaskruchten) [inline src/components/fields/derived.js]: what is this ?

Comment (nicolaskruchten) [reply src/components/fields/derived.js]: Oh i see, nevermind ;)

Comment (dmt0) [reply src/components/fields/derived.js]: Actually, "what is this" is a very relevant question :) 
I reread the docs and see that it's not needed. zmin and zmax do not depend on zauto - I'm removing this.

**Pull request merged**